lang/gcc10-bootstrap/Portfile

```diff
@@ -169,61 +161,124 @@ platform darwin 8 {
         ln -s ${workpath}/${MAKE_V}/make   ${workpath}/bins/make
     }
 
     configure.env-append    MAKE=${workpath}/bins/make
     build.env-append        MAKE=${workpath}/bins/make
-}
 
-set prefix          ${prefix}/libexec/${name}
+    configure.env-replace   PATH=/usr/bin:/bin:/usr/sbin:/sbin \
+                            PATH=${workpath}/bins:/usr/bin:/bin:/usr/sbin:/sbin
+
+    build.env-replace       PATH=/usr/bin:/bin:/usr/sbin:/sbin \
+                            PATH=${workpath}/bins:/usr/bin:/bin:/usr/sbin:/sbin
+}
 
 configure.cmd       ${worksrcpath}/configure
 configure.dir       ${workpath}/build
+configure.pre_args  --prefix=${prefix}/libexec/${name}
 configure.args      --enable-languages=c,c++,objc,obj-c++ \
                     --enable-bootstrap \
                     --with-system-zlib \
+                    --with-local-prefix=${prefix}/libexec/${name} \
                     --disable-nls \
                     --disable-tls \
+                    --disable-multilib \
+                    --disable-multiarch \
                     --with-bugurl=https://trac.macports.org/newticket
 
 if {${os.major} >= 18 && ${configure.sdkroot} ne ""} {
     configure.args-append --with-sysroot="[regsub {MacOSX1[0-9]\.[0-9]+\.sdk} ${configure.sdkroot} {MacOSX.sdk}]"
 }
 
-if {${os.major} < 18} {
-    universal_variant yes
-    default_variants +universal
-} else {
-    universal_variant no
-    configure.args-append --disable-multilib
+set universal_targets ""
+foreach arch ${configure.universal_archs} {
+    switch -- ${arch} {
+        arm64   {set universal_targets "${universal_targets},aarch64-apple-${os.platform}${os.version}"}
+        x86_64  {set universal_targets "${universal_targets},x86_64-apple-${os.platform}${os.version}"}
+        i386    {set universal_targets "${universal_targets},i686-apple-${os.platform}${os.version}"}
+        ppc     {set universal_targets "${universal_targets},powerpc-apple-${os.platform}${os.version}"}
+        ppc64   {set universal_targets "${universal_targets},powerpc64-apple-${os.platform}${os.version}"}
+    }
 }
 
-platform powerpc {
-    configure.universal_archs ppc ppc64
-}
-platform i386 {
-    configure.universal_archs i386 x86_64
-}
-platform arm {
-    configure.universal_archs x86_64 arm64
+variant universal {
+    configure.args-replace \
+                    --disable-multilib \
+                    --enable-multilib
+
+    configure.args-replace \
+                    --disable-multiarch \
+                    --enable-multiarch
+
+    if { [string length ${universal_targets}] > 0 } {
+        configure.args-append \
+            --enable-targets=[string replace ${universal_targets} 0 0]
+    }
 }
 
+default_variants    +universal
+
 platform darwin 8 {
-    configure.args-append --with-dwarf2
+    configure.args-append \
+                    --with-dwarf2
+}
+
+merger_arch_flag            yes
+merger_arch_compiler        yes
+merger_must_run_binaries    yes
```

Comment:
must it run binaries? That limits the universal PortGroup somewhat, which is OK, if it must be. gcc is such a robust build/host/target system... what happened without this?

Reply:
I was thinking that also as I commonly run port under Rosetta2 on my M1 Mac

Reply:
@kencu yep, to make a full bootstrap of gcc, and not just cross compile it. Cross compile to target requires to build properly gmp and stage0 compiler, which is quite tricky and for some combination it will fail.

We may try to remove this limitation as the next step.